pkg/util/overridemanager/overridemanager.go

```diff
@@ -136,54 +137,42 @@ func (o *overrideManagerImpl) applyNamespacedOverrides(rawObj *unstructured.Unst
 	return appliedList, nil
 }
 
-func (o *overrideManagerImpl) getMatchedClusterOverridePolicy(policies []policyv1alpha1.ClusterOverridePolicy, resource *unstructured.Unstructured, cluster *clusterv1alpha1.Cluster) []policyv1alpha1.ClusterOverridePolicy {
-	// select policy in which at least one resource selector matches target resource.
-	resourceMatches := make([]policyv1alpha1.ClusterOverridePolicy, 0)
+func (o *overrideManagerImpl) getMatchingClusterOverridePolicies(policies []policyv1alpha1.ClusterOverridePolicy, resource *unstructured.Unstructured, cluster *clusterv1alpha1.Cluster) []policyv1alpha1.ClusterOverridePolicy {
+	matchingPolicies := make([]policyv1alpha1.ClusterOverridePolicy, 0)
 	for _, policy := range policies {
-		if OverridePolicyMatches(resource, policy.Spec.ResourceSelectors) {
-			resourceMatches = append(resourceMatches, policy)
-		}
-	}
-
-	// select policy which cluster selector matches target resource.
-	clusterMatches := make([]policyv1alpha1.ClusterOverridePolicy, 0)
-	for _, policy := range resourceMatches {
-		if util.ClusterMatches(cluster, policy.Spec.TargetCluster) {
-			clusterMatches = append(clusterMatches, policy)
+		if util.ResourceMatchSelectors(resource, policy.Spec.ResourceSelectors...) &&
```

Comment:
For current implementation `util.ResourceMatchSelectors()` will return false when `policy.Spec.ResourceSelectors == nil`.

How about fix it as you have done in https://github.com/karmada-io/karmada/blob/d91f7868d066af6edd54b1cc55bacbd9bc6aaf55/pkg/util/detector/detector.go#L271-L274 ?

Reply:
I added a logic or condition:
```go
if (policy.Spec.ResourceSelectors == nil || util.ResourceMatchSelectors(resource, policy.Spec.ResourceSelectors...)) &&
			(policy.Spec.TargetCluster == nil || util.ClusterMatches(cluster, policy.Spec.TargetCluster)) {
			...
		}
```
Do you think that's okay?

Reply:
Done